### ISSUE TITLE
ReflectionUtils.getDeclaredMethods fail when use jdk 21: Unnamed Variables and Patterns 

### ISSUE THREAD

**Issue by billschen** [spring-projects/spring-framework#31392]
**Affects:** \<Spring Framework version>

---

Affects:spring-core-6.0.12.jar:6.0.12
I use a simple enum like
```java
import java.util.function.Function;
import java.math.BigDecimal;
public enum EchoMessage {
    normal(BigDecimal::toString),
    plain(BigDecimal::toPlainString),
    fixed(_-> "hhhh"); // Unnamed Variables and Patterns 
    public final Function<BigDecimal, String> toMsg;
    EchoMessage(Function<BigDecimal, String> toMsg){
        this.toMsg = toMsg;
    }
}

// and a controller like:


@GetMapping(value = "/echo-message")
    public List<String> getEchoMessages( ){
        return Arrays.stream( EchoMessage.values() ).map( e->e.toMsg.apply(new BigDecimal("54136.5412"))).collect(Collectors.toList());
    }
```

exception happen:
Caused by: java.lang.ClassFormatError: **Illegal field name ""** in class com/kbnotebooks/annotation/creator/service/echo/EchoMessage
	at java.base/java.lang.ClassLoader.defineClass1(Native Method) ~[na:na]
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1027) ~[na:na]
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[na:na]
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862) ~[na:na]
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760) ~[na:na]
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681) ~[na:na]
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[na:na]
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method) ~[na:na]
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3578) ~[na:na]
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2676) ~[na:na]
	at org.springframework.util.**ReflectionUtils.getDeclaredMethods**(ReflectionUtils.java:465) ~[spring-core-6.0.12.jar:6.0.12]

when _not use_ Unnamed Variables and Patterns, exception _not happen_

**Comment by mdeinum** [comment]
This is a bug in the JDK as reported [here](https://bugs.openjdk.org/browse/JDK-8313323?jql=issuetype%20%3D%20Bug%20AND%20text%20~%20%22unnamed%20variable%22). The fix is backported to Java 21.0.1 (so the next fix release).

**Comment by billschen** [comment]
> This is a, unfortunate, bug in the JDK as reported [here](https://bugs.openjdk.org/browse/JDK-8313323?jql=issuetype%20%3D%20Bug%20AND%20text%20~%20%22unnamed%20variable%22). The fix is backported to Java 21.0.1 (so the next fix release).

ok,thank you,Sorry,I use javac with out -g option and  none ClassFormatError error.so I think is ReflectionUtils bug.
